datafusion/proto/src/lib.rs

```diff
@@ -15,8 +15,96 @@
 // specific language governing permissions and limitations
 // under the License.
 
-//! Serde code for logical plans and expressions.
-
+//! Serialize / Deserialize DataFusion Plans to bytes
+//!
+//! This crate provides support for serializing and deserializing the
+//! following structures to and from bytes:
+//!
+//! 1. [`LogicalPlan`]'s (including [`Expr`]),
+//! 2. [`ExecutionPlan`]s (including [`PhysicalExpr`])
+//!
+//! [`LogicalPlan`]: datafusion_expr::LogicalPlan
+//! [`Expr`]: datafusion_expr::Expr
+//! [`ExecutionPlan`]: datafusion::physical_plan::ExecutionPlan
+//! [`PhysicalExpr`]: datafusion::physical_expr::PhysicalExpr
+//!
+//! Internally, this crate is implemented by converting the plans to [protocol
+//! buffers] using [prost].
+//!
+//! [protocol buffers]: https://developers.google.com/protocol-buffers
+//! [prost]: https://docs.rs/prost/latest/prost/
+//!
+//! # See Also
+//!
+//! The binary format created by this crate supports the full range of DataFusion
+//! plans, but is DataFusion specific. See [datafusion-substrait] for a crate
+//! which can encode many DataFusion plans using the [substrait.io] standard.
```

Comment:
Since format is DataFusion specific, are we able to give any info on version compatibility here, etc.? If that has been decided yet :thinking:

Reply:
Good call -- I think the current state of affairs is that there is no gaurantee about version compatibility, and I added that in 7d91abf8e. If version compatibility is a desired feature I think what is most important would be a test harness for verifying it (which is a non trivial undertaking)